### PR TITLE
faster Pipeline Jenkins, by merging Check- and Build-Stage

### PR DESCRIPTION
**Pull request by fel115**
This PR is connected with the issue #241. I created this PR to test the created Pipeline on our Jenkins instance. So for now it isn't ready to merge. 

## Basics

- [x] Details of what you changed are in commit messages.
- [x] References to issues, e.g. `close #X`, are in the commit messages.
- [x] The buildserver is happy.
- [x] The PR is rebased with current master.

## Checklist

- [ ] I fully described what my PR does in the documentation
      (not in the PR description)
- [ ] I fixed all affected documentation
- [ ] I fixed all affected decisions
- [ ] I added unit tests for my code
- [ ] I added code comments, logging, and assertions as appropriate
- [ ] I translated all strings visible to the user
- [ ] I mentioned [every code or binary](/.reuse/dep5) not directly written or done by me in [reuse syntax](https://reuse.software/)
- [ ] I created left-over issues for things that are still to be done
- [ ] Code is conforming to [our Architecture](/doc/architecture)
- [ ] Code is conforming to [our Guidelines](/doc/guidelines)
      (exceptions are documented)
- [ ] Code is consistent to [our Design Decisions](/doc/decisions)

## Review

- [ ] I've tested the code
- [ ] I've read through the whole code
- [ ] Examples are well chosen and understandable

### CODE REVIEW

**Comment by lukashartl** [comment]
pls rebase onto latest master 😺

**Comment by fel115** [comment]
@lukashartl I think this Pipeline runs now faster, because there are less dependencies loaded twice. Maybe you could review my changes?

File: ci/Jenkinsfile [hunk 1]
```diff
@@ -50,63 +50,105 @@ stage('Checks') {
...
-                    sh "npm run format:check"
-                    sh "npm run lint"
-                    sh "npm run test"
+                stage('build-frontend'){
+                    sh './ci/build-scripts/build-frontend.sh'
+                    stash includes: 'frontend/dist/**/*', name: 'frontend'
                 }
             }
+
+            /*stage('build-frontend'){
```
Comment (markus2330) [inline ci/Jenkinsfile:143]: Is this commented out on purpose?

File: ci/Jenkinsfile [hunk 2]
```diff
@@ -115,7 +157,7 @@ stage('Checks') {
 
 // REQUIRED for Deployment
 // see Jenkinsfile.release
-stage('Build in Docker'){
+/*stage('Build in Docker'){
```
Comment (markus2330) [inline ci/Jenkinsfile:160]: Better remove if it is not needed anymore.

**Review by markus2330 — CHANGES_REQUESTED** [review]
Great work. Can you maybe tell us the changes in the build time?

**Comment by fel115** [comment]
The time improvement is due to the onetime build of the docker image and the cached build dep in the docker image. So we don't have to do some tasks twice, like downloading deps.

**Review by markus2330 — APPROVED** [review]
LGTM, @lukashartl please also have a look if it doesn't have unintended changes to deployment.

**Comment by markus2330** [comment]
We should merge this asap, I've put the urgent label.

**Review by lukashartl — APPROVED** [review]
LGTM.
And since PR deploy works, DEV deploy should also work :rocket:

**Pull request merged**